sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2889,6 +2889,24 @@ class Dataset[T] private[sql](
     flatMap(func)(encoder)
   }
 
+  /**
+   * Return a new Dataset of string created by piping elements to a forked external process.
+   * The resulting Dataset is computed by executing the given process once per partition.
+   * All elements of each input partition are written to a process's stdin as lines of input
+   * separated by a newline. The resulting partition consists of the process's stdout output, with
+   * each line of stdout resulting in one element of the output partition. A process is invoked
+   * even for empty partitions.
```

Comment:
This sounds like risky to me, `A process is invoked even for empty partitions`.
This may cause a hang situation if the command is expecting input.
For example, this PR's test case is using `cat`. And, `cat | wc -l` hangs.
Could you add a test case of empty partition to make it sure that we handle those cases?

Reply:
Otherwise, you may want to add some warnings here.

Reply:
This is a good point. As you see this uses RDD.pipe, let me check if it addresses this or not. And yeah I will add a test case too.